examples/runtime_torture/launchable_torture.py

```diff
@@ -0,0 +1,95 @@
+# Copyright 2024 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+import argparse
+from contextlib import contextmanager
+import sys
+from time import perf_counter
+
+import torch
+import torch.nn as nn
+
+import shark_turbine.aot as aot
+
+from shark_turbine.runtime import (
+    Launchable,
+)
+
+
+@contextmanager
+def report_time(banner):
+    t1 = perf_counter()
+    yield
+    t2 = perf_counter()
+    print(f"{banner}: {(t2 - t1) * 1000}ms")
+
+
+class TestLinearModel(nn.Module):
+    def forward(self, input, weight, bias):
+        result = nn.functional.linear(input, weight, bias)
+        scale = 1 / 256.0
+        result = result * scale
+        return nn.functional.tanh(result)
+
+
+def test_sequential(nreps, device, dtype):
+    print("*** run test_sequential:")
+    print("------------------------")
+    print("*** compiling model")
+    torch.manual_seed(42)
+    initial_input = torch.rand([512, 512], dtype=dtype)
+    weight = torch.rand([512, 512], dtype=dtype)
+    bias = torch.rand([512], dtype=dtype)
+    eo = aot.export(TestLinearModel(), args=(initial_input, weight, bias))
+    launcher = Launchable.jit_compile(str(eo.mlir_module))
```

Comment:
For my own understanding--isn't AOT in the above enough? what we are jitting here for? (I can read the code later if it's too much to type out.)

Reply:
Exporting just does a torch.export to IR. Setting up the launchable in this way says that it should JIT compile from the source IR for every device it is launched on (i.e. I have three GPUs of two different architectures and if I run this on different ones, it will on the fly compile for that arch). The `preload` call just below, though, says to compile synchronously for the given device.

Ideally this is all wrapped in a higher level API at some point. But for now, it is explicit.

Reply:
I see. Thanks for the explanation!